File: DSA/akashyadav_590014126/Day_41_Question_2.py
def decode_string(s):
    count_stack=[]
    string_stack=[]
    current_str=' '
    k=0

    for char in s:
        if char.isdigit():
            k=k*10 + int(char)
        elif char == '[':
            count_stack.append(k)
            string_stack.append(current_str)
            current_str=''
            k=0
        elif char == ']':
            repeat=count_stack.pop()
            prev_str=string_stack.pop()
            current_str=prev_str + current_str*repeat
        else:
            current_str += char

    return current_str

#example
print(decode_string("3[a]2[bc]"))

File: DSA/VaniJain_590012020/Day41Question2.java
import java.util.*;
public class DecodeString 
{
    public static String decodeString(String s) 
  {
        Stack<Integer> countStack = new Stack<>();
        Stack<StringBuilder> stringStack = new Stack<>();
        StringBuilder currentStr = new StringBuilder();
        int k = 0;
        for (char ch : s.toCharArray()) 
        {
            if (Character.isDigit(ch)) 
            {
                k = k * 10 + (ch - '0');
            } else if (ch == '[') 
            {
                countStack.push(k);
                stringStack.push(currentStr);
                k = 0;
                currentStr = new StringBuilder();
            } else if (ch == ']') 
            {
                int repeatCount = countStack.pop();
                StringBuilder prevStr = stringStack.pop();
                StringBuilder temp = new StringBuilder(prevStr);
                for (int i = 0; i < repeatCount; i++) {
                    temp.append(currentStr);
                }
                currentStr = temp;
            } 
            else 
            {
                currentStr.append(ch);
            }
        }
        return currentStr.toString();
    }
    public static void main(String[] args) 
  {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter encoded string (e.g., 3[a]2[bc]): ");
        String input = scanner.nextLine();
        String decoded = decodeString(input);
        System.out.println("Decoded string: " + decoded);
    }
}
